salt/metalk8s/orchestrate/bootstrap/init.sls

```diff
@@ -1,23 +1,32 @@
 {# Because of the grain lookup below, the bootstrap minion *must* be available
    before invoking this SLS, otherwise rendering will fail #}
-{%- set bootstrap_control_plane_ip = salt.saltutil.cmd(
-        tgt=pillar.bootstrap_id,
-        fun='grains.get',
-        kwarg={
-            'key': 'metalk8s:control_plane_ip',
-        },
-    )[pillar.bootstrap_id]['ret']
-%}
-
-{%- set bootstrap_workload_plane_ip = salt.saltutil.cmd(
-        tgt=pillar.bootstrap_id,
-        fun='grains.get',
-        kwarg={
-            'key': 'metalk8s:workload_plane_ip',
-        },
-    )[pillar.bootstrap_id]['ret']
-%}
 
+{%- set max_try = 5 %}
+{%- set bootstrap_grains = {} %}
+{%- for _ in range(max_try) %}
```

Comment:
BTW, could be interesting to add some unit test cases for this logic (would need to customize the mocks to simulate "intermittent failures" though).

Reply:
As discussed offline with @alexandre-allard-scality it will be handle in another PR since those tests do not exists in development/2.8